### PR TITLE
fix(kuma-cp) CNI relies on annotations

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Summary

The current state of CNI has a couple of problems after recent changes

* On the latest OpenShift namespace name for controller was empty (due to Watch on the Mesh)
* RBAC for NAD was lost when migrating to HELM
* Namespace controller was relying on labels, not annotations
* We were not deleting NAD when a user removes Kuma annotation from the namespace

### Documentation

- [ ] No docs, internal fixes.

### CODE REVIEW

**Review by nickolaev — APPROVED** [review]
Looks good to me.

File: pkg/plugins/runtime/k8s/controllers/namespace_controller.go [hunk 1]
```diff
@@ -29,11 +24,8 @@ type NamespaceReconciler struct {
 	kube_client.Client
 	Log logr.Logger
 
-	SystemNamespace         string
-	CNIEnabled              bool
-	SkipDefaultMeshCreation bool
-	ResourceManager         core_manager.ResourceManager
-	DefaultMeshTemplate     mesh_proto.Mesh
+	CNIEnabled      bool
+	ResourceManager core_manager.ResourceManager
```
Comment (lobkovilya) [inline pkg/plugins/runtime/k8s/controllers/namespace_controller.go:28]: Is it used?

**Review by lobkovilya — APPROVED** [review]
`ResourceManager` looks like an unused variable. Other than that approve

File: pkg/plugins/runtime/k8s/controllers/namespace_controller.go (hunk 1 above)
Comment (jakubdyszkiewicz) [reply pkg/plugins/runtime/k8s/controllers/namespace_controller.go:28]: I'll fix by the way it in the next PR that I'll be introducing

**Pull request merged**